File: 0901-online-stock-span/0901-online-stock-span.cpp
class StockSpanner {
public: 
    std::vector<int> history;
    
    StockSpanner() {
        
    }
    
    int next(int price) {
        history.push_back(price);
        
        for(auto iter=history.rbegin(); iter != history.rend(); ++iter) {
            if (*iter > price) {
                return std::distance(history.rbegin(), iter);
            }
        }
        return history.size();
    }
        
        // auto result = std::find_if(history.rbegin(), history.rend(),
        //                            [](int i, int price) { return i > price; });
        // return std::distance(result, history.rbegin());
};

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner* obj = new StockSpanner();
 * int param_1 = obj->next(price);
 */

File: Medium/0901-online-stock-span/0901-online-stock-span.py
class StockSpanner:

    def __init__(self):
        import numpy as np
        self.nums = np.array([])

    def next(self, price: int) -> int:
        import numpy as np
        self.nums = np.append(self.nums, int(price))
        # print(self.nums)
        try:
            days = self.nums.shape[0] - np.argwhere(self.nums > price).flatten()[-1] - 1
            # print(np.argwhere(self.nums > price))
            # print("a")
        except:
            days = self.nums.shape[0]
            # print("b")
        # print("*"*10)
        return days


# Your StockSpanner object will be instantiated and called as such:
# obj = StockSpanner()
# param_1 = obj.next(price)